scripts/update-dependencies/Program.cs

```diff
@@ -1,68 +1,76 @@
-        private static bool ParseArgs(string[] args)
+        private static async Task CreatePullRequest()
         {
-            foreach (string arg in args)
+            var gitHubAuth = new GitHubAuth(config.GithubToken, config.GithubUserName, config.GithubEmail);
+            var origin = new GitHubProject(config.GitHubProject, config.GithubUserName);
+            var upstreamBranch = new GitHubBranch(config.GitHubUpstreamBranch, new GitHubProject(config.GitHubProject, config.GitHubUpstreamOwner));
+
+            var commitMessage = $"Updating external dependencies to '{ await GetOrchestratedBuildId() }'";
+            var body = string.Empty;
+            if (config.GitHubPullRequestNotifications.Any())
             {
-                int idx = arg.IndexOf('=');
-                if (idx < 0)
-                {
-                    Console.Error.WriteLine($"Unrecognized argument '{arg}'");
-                    return false;
-                }
+                body += PullRequestCreator.NotificationString(config.GitHubPullRequestNotifications);
+            }
 
-                string name = arg.Substring(0, idx);
-                string value = arg.Substring(idx + 1);
-                Environment.SetEnvironmentVariable(name, value);
+            foreach (var updatedVersion in Config.Instance.UpdatedVersions)
+            {
+                body += $"{updatedVersion}{Environment.NewLine}";
             }
+            Console.WriteLine($"Title: " + commitMessage);
+            Console.WriteLine($"Body: " + body);
 
-            return true;
+            await new PullRequestCreator(gitHubAuth, origin, upstreamBranch)
+                .CreateOrUpdateAsync(commitMessage, commitMessage + $" ({upstreamBranch.Name})", body);
         }
 
-        private static void CreatePullRequest()
+        private static async Task<string> GetOrchestratedBuildId()
         {
-            GitHubAuth gitHubAuth = new GitHubAuth(config.Password, config.UserName, config.Email);
-            GitHubProject origin = new GitHubProject(config.GitHubProject, config.UserName);
-            GitHubBranch upstreamBranch = new GitHubBranch(config.GitHubUpstreamBranch, new GitHubProject(config.GitHubProject, config.GitHubUpstreamOwner));
+            var xmlUrl = Config.Instance.BuildXml;
 
-            string commitMessage = "Updating external dependencies";
-            string body = string.Empty;
-            if (config.GitHubPullRequestNotifications.Any())
+            using (var client = new HttpClient())
             {
-                body += PullRequestCreator.NotificationString(config.GitHubPullRequestNotifications);
+                var response = await client.GetAsync(xmlUrl);
+                using (var bodyStream = await response.Content.ReadAsStreamAsync())
+                {
+                    var xmlDoc = new XmlDocument();
+                    xmlDoc.Load(bodyStream);
+                    var orcBuild = xmlDoc.GetElementsByTagName("OrchestratedBuild")[0];
```

Comment:
We should do some error checking here. This will fail with an out of bound index if the tag does not exist which wouldn't be helpful. Have this print the contents of the xml if the tag is empty or missing BuildId.